web/modules/custom/asset_release/tests/src/Kernel/AssetReleaseWorkflowTest.php

```diff
@@ -0,0 +1,385 @@
+use Drupal\og\Entity\OgRole;
+use Drupal\rdf_entity\Entity\Rdf;
+use Drupal\rdf_entity\RdfInterface;
+use Drupal\Tests\joinup_core\JoinupWorkflowTestBase;
+
+/**
+ * Tests the support of saving various encoded stings in the triple store.
+ *
+ * @group asset_release
+ */
+class AssetReleaseWorkflowTest extends JoinupWorkflowTestBase {
+
+  /**
+   * A user with the authenticated role.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $userAuthenticated;
+
+  /**
+   * A user with the moderator role.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $userModerator;
+
+  /**
+   * A user to be used as a solution facilitator.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $userOgFacilitator;
+
+  /**
+   * A user to be used as a solution administrator.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $userOgAdministrator;
+
+  /**
+   * The solution parent entity.
+   *
+   * @var \Drupal\rdf_entity\RdfInterface
+   */
+  protected $solutionGroup;
+
+  /**
+   * The solution facilitator role.
+   *
+   * @var \Drupal\og\Entity\OgRole
+   */
+  protected $roleFacilitator;
+
+  /**
+   * The solution administrator role.
+   *
+   * @var \Drupal\og\Entity\OgRole
+   */
+  protected $roleAdministrator;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function setUp() {
+    parent::setUp();
+
+    $this->userAuthenticated = $this->createUserWithRoles();
+    $this->userModerator = $this->createUserWithRoles(['moderator']);
+    $this->userOgFacilitator = $this->createUserWithRoles();
+    $this->userOgAdministrator = $this->createUserWithRoles();
+
+    $this->roleFacilitator = OgRole::getRole('rdf_entity', 'solution', 'facilitator');
+    $this->roleAdministrator = OgRole::getRole('rdf_entity', 'solution', 'administrator');
+  }
+
+  /**
+   * Creates a user with roles.
+   *
+   * @param array $roles
+   *    An array of roles to initialize the user with.
+   *
+   * @return \Drupal\Core\Session\AccountInterface
+   *    The created user object.
+   */
+  public function createUserWithRoles($roles = []) {
+    $user = $this->createUser();
+    foreach ($roles as $role) {
+      $user->addRole($role);
+    }
+    $user->save();
+
+    return $user;
+  }
+
+  /**
+   * Tests the CRUD operations for the asset release entities.
+   */
+  public function testCrudAccess() {
+    // Test create access.
+    foreach ($this->createAccessProvider() as $parent_state => $test_data_arrays) {
+      $parent = $this->createDefaultParent($parent_state);
+      $content = Rdf::create([
+        'rid' => 'asset_release',
+        'field_isr_is_version_of' => $parent->id(),
+      ]);
+      foreach ($test_data_arrays as $test_data) {
+        $operation = 'create';
+        $user_var = $test_data[0];
+        $expected_result = $test_data[1];
+
+        $access = $this->ogAccess->userAccessEntity('create', $content, $this->{$user_var})->isAllowed();
```

Comment:
Probably I missexplained the issue in the previous call. I do not think that the og is meant to handle the entity create access by hooking into that. The issue is simple:
The parent entity is not passed in the entity access controller and we have no way of knowing it.

The call stack is:
EntityAccessControllHandler::createAccess invokes hook_entity_create_access
og_entity_create_access

The only context we have here is the entity type id. We cannot rely on a context to determine the parent or the field itself as it is a new entity.
This is why we went with directly calling the ogAccess service.
The og_entity_create_access offers some help and convenience mainly for node.
@See og.module::og_entity_create_access.